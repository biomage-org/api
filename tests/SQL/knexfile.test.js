const config = require('../../src/config');
const getConnectionParams = require('../../src/SQL/getConnectionParams');

const knexfile = require('../../src/SQL/knexfile');

jest.mock('../../src/config');

const mockConnectionParams = { fake: true };
jest.mock('../../src/SQL/getConnectionParams', () => jest.fn(() => mockConnectionParams));

describe('knexfile', () => {
  it('Works correctly for staging', async () => {
    config.clusterEnv = 'staging';

    const res = await knexfile();

    expect(getConnectionParams).toHaveBeenCalled();

    // @ts-ignore
    expect(res.staging).toBeTruthy();

    // @ts-ignore
    expect(res.staging.connection).toEqual(mockConnectionParams);

    expect(res).toMatchSnapshot();
  });
});
