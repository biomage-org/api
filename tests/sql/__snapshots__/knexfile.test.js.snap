// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`knexfile Works correctly for staging 1`] = `
Object {
  "staging": Object {
    "client": "postgresql",
    "connection": Object {
      "fake": true,
    },
    "postProcessResponse": [Function],
  },
}
`;
